src/popover/popover.ts

```diff
@@ -208,13 +207,17 @@ export class NgbPopover implements OnInit, OnDestroy, OnChanges {
 
       this._renderer.setAttribute(this._elementRef.nativeElement, 'aria-describedby', this._ngbPopoverWindowId);
 
-      if (this.container === 'body') {
+      const isBodyContainer = this.container === 'body';
+      if (isBodyContainer) {
         this._document.querySelector(this.container).appendChild(this._windowRef.location.nativeElement);
       }
 
       // apply styling to set basic css-classes on target element, before going for positioning
       this._windowRef.changeDetectorRef.markForCheck();
 
+      if (isBodyContainer) {
```

Comment:
could we simply merge this condition into the previous one? (same for the tooltip)

```ts
if (this.container === 'body') {
    this._document.querySelector(this.container).appendChild(this._windowRef.location.nativeElement);
    ngbScroll(this._ngZone, this._elementRef.nativeElement, () => this.positionElements(), this.hidden);
}
```